fortnite_api/enums.py

```diff
@@ -47,7 +49,133 @@
 )
 
 
-class KeyFormat(enum.Enum):
+E = TypeVar('E', bound='Enum')
+
+
+def _create_value_cls(name: str, comparable: bool):
+    # All the type ignores here are due to the type checker being unable to recognise
+    # Runtime type creation without exploding.
+    cls = namedtuple('_EnumValue_' + name, 'name value')
+    cls.__repr__ = lambda self: f'<{name}.{self.name}: {self.value!r}>'  # type: ignore
+    cls.__str__ = lambda self: f'{name}.{self.name}'  # type: ignore
+    if comparable:
+        cls.__le__ = lambda self, other: isinstance(other, self.__class__) and self.value <= other.value  # type: ignore
+        cls.__ge__ = lambda self, other: isinstance(other, self.__class__) and self.value >= other.value  # type: ignore
+        cls.__lt__ = lambda self, other: isinstance(other, self.__class__) and self.value < other.value  # type: ignore
+        cls.__gt__ = lambda self, other: isinstance(other, self.__class__) and self.value > other.value  # type: ignore
+    return cls
+
+
+def _is_descriptor(obj):
+    return hasattr(obj, '__get__') or hasattr(obj, '__set__') or hasattr(obj, '__delete__')
+
+
+class EnumMeta(type):
+    if TYPE_CHECKING:
+        __name__: ClassVar[str]
+        _enum_member_names_: ClassVar[List[str]]
+        _enum_member_map_: ClassVar[Dict[str, Any]]
+        _enum_value_map_: ClassVar[Dict[Any, Any]]
+
+    def __new__(
+        cls,
+        name: str,
+        bases: Tuple[type, ...],
+        attrs: Dict[str, Any],
+        *,
+        comparable: bool = False,
+    ) -> EnumMeta:
+        value_mapping = {}
+        member_mapping = {}
+        member_names = []
+
```

Comment:
Missing definition annotations, causing every reference to these to be unknown.

```suggestion
        value_mapping: Dict[OldValue, NewValue] = {}
        member_mapping: Dict[str, NewValue] = {}
        member_names: List[str] = []

```
Where `OldValue` and `NewValue` are typealiases that simply point to `Any`

Reply:
Is pyright picking up the `pyproject.toml` on your local development environment? This raised multiple type erros for me.

Reply:
I have to admit that Pyright is currently disabled in my IDE, hence the plugin isn't updated for the latest PyCharm version 🥲